src/setup_payload/Base38Encode.cpp

```diff
@@ -81,6 +81,8 @@ CHIP_ERROR base38Encode(ByteSpan in_buf, MutableCharSpan & out_buf)
     if (out_idx < out_buf.size())
     {
         out_buf.data()[out_idx] = '\0';
+        // Reduce output span size to not include null-terminator.
+        out_buf.reduce_size(out_buf.size() - 1);
```

Comment:
```suggestion
        out_buf.reduce_size(out_idx);
```
Because we have written `out_idx` bytes into there, right?  The reducing to "total size of buffer - 1" would still leave garbage in the span if the data is not as big as it can get.